libraries/chain/include/eosio/chain/hotstuff/hotstuff.hpp

```diff
@@ -211,6 +212,8 @@ namespace eosio::chain {
       bool is_weak()   const { return !!_weak_votes; }
       bool is_strong() const { return !_weak_votes; }
 
+      uint32_t accumulated_weight() const { return (_strong_votes ? _strong_votes.value().count() : 0) + (_weak_votes ? _weak_votes.value().count() : 0); }
+
```

Comment:
I believe the `accumulated_weight` function should not count the weak votes when the qc is strong.

Reply:
When qc is strong, weak count is 0. No need to have a separate expression. Will add a comment.

Reply:
Curious why your preference for `->` and `*` over `.value()` ?

Reply:
`value()` is clearer while `->` is shorter? Either way is fine with me.

Reply:
> Curious why your preference for `->` and `*` over `.value()` ?

I believe it is simpler, more compact, and more idiomatic. For example the example in cppreference uses `*` and not `value()`. Also it looks better to me because it works just like a pointer, and I can tell with a quick look that it is correct, while the `value()` version takes more effort to read.

I guess it is a matter of preference. Personally, everything else (like legibility) being equal, I prefer the most concise way of expressing something.

Reply:
Oh and I forgot `*` could actually be more efficient because it doesn't check that the optional contains a value (while `value()` throws if it doesn't).